### PR TITLE
feat: switch focus on shift + tab key press

### LINKED ISSUE #130: Log table might violate Constraint-Typing-Preferred
**Describe the bug**
There is currently no way to navigate the list of logs other than using a mouse to scroll, this might be considered a violation of the Constraint-Typing-Preferred constraint.

We can table this for discussion

### PR DESCRIPTION
**Pull request by SuhailLoya**
fixes #130 

Changes made
1) Pressing shift+tab will cycle focus between command box, log list(if available) and person list pane
2) whenever a selected index is returned from command result, now the corresponding person card will be highlighted in person list pane
3) Arrow keys scrolling in person list panel now also updates the person detail card on the right pane

### CODE REVIEW

File: src/main/java/careconnect/ui/PersonListPanel.java [hunk 1]
```diff
@@ -24,20 +24,25 @@ public class PersonListPanel extends UiPart<Region> {
     /**
      * Creates a {@code PersonListPanel} with the given {@code ObservableList}.
      */
-    public PersonListPanel(ObservableList<Person> personList, Consumer<Integer> showSelectedPerson) {
+    public PersonListPanel(ObservableList<Person> personList, Consumer<Integer> setAndShowSelectedPerson) {
         super(FXML);
         personListView.setItems(personList);
```
Comment (AppleJem) [inline src/main/java/careconnect/ui/PersonListPanel.java:29]: Do we need to add `this.personListView...` ? Cuz there was `this` in the other uses of `personListView

**Review by AppleJem — APPROVED** [review]
Goddam bro you're insane

and tiny weenie nitpick

File: src/main/java/careconnect/ui/ShiftTabFocusable.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package careconnect.ui;
+
+/**
+ * interface for focusable ui components
+ */
+public interface ShiftTabFocusable {
```
Comment (mongj) [inline src/main/java/careconnect/ui/ShiftTabFocusable.java:6]: very neat design!

File: src/main/java/careconnect/ui/MainWindow.java [hunk 3]
```diff
@@ -65,14 +72,54 @@ public MainWindow(Stage primaryStage, Logic logic) {
...
         setAccelerators();
     }
 
+    /**
+     * Updates current focus item to provided focus item.
+     *
+     * @param focusItem the focusItem to update
+     */
+    public void updateCurrentFocusItem(FocusItems focusItem) {
+        assert(focusItem != null);
```
Comment (mongj) [inline src/main/java/careconnect/ui/MainWindow.java:92]: good use of defensive coding!

**Review by mongj — COMMENTED** [review]
thanks for implementing this! just need to fix the checkstyle issues and we're good to go :)

**Pull request merged**